include/net/net_offload.h

```diff
@@ -420,6 +422,88 @@ static inline int net_offload_put(struct net_if *iface,
 	return net_if_offload(iface)->put(context);
 }
 
+#else
+static inline int net_offload_get(struct net_if *iface,
```

Comment:
This cannot be done like this. The inline version must be placed together with the documentation block so that documentation is generated properly.

Reply:
let's try with @cond INTERNAL_HIDDEN

Reply:
Then we do not get any documentation at all for these functions. We need to follow the documentation pattern like this
```
/**
 * @brief Free/close a network context.
 *
 * @details This releases the context. It is not possible to
 * send or receive data via this context after this call.
 * This is similar as BSD shutdown() function.
 *
 * @param iface Network interface where the offloaded IP stack can be
 * reached.
 * @param context The context to be closed.
 *
 * @return 0 if ok, < 0 if error
 */
static inline int net_offload_put(struct net_if *iface,
				  struct net_context *context)
{
#if defined(CONFIG_NET_OFFLOAD)
	NET_ASSERT(iface);
	NET_ASSERT(net_if_offload(iface));
	NET_ASSERT(net_if_offload(iface)->put);
	return net_if_offload(iface)->put(context);
#else 
	return 0;
#endif 
}
```
This is very ugly looking but needed because of limitations in doxygen. I just do not want to fix this one later when we generate documentation for 2.0.